### ISSUE TITLE
Inconsistent Results: Wappalyzer Detects Different Technologies on Each Scan for "https://www.clairestudio.co.uk/" 

### ISSUE THREAD

**Issue by kundan8239** [wappalyzer/wappalyzer#7893]
* Description*
it consistently provides varying results for the domain `https://www.clairestudio.co.uk` upon each scan. The problem lies in the fact that the detected technologies do not remain consistent across consecutive scans, leading confusion

*Steps to Reproduce*
1. Input the URL "https://www.clairestudio.co.uk/" for scanning.
2. Record the detected technologies.
3. Repeat the scan process multiple times, preferably at different intervals or using different instances of the tool.

*Additional Information*
The issue has been observed across multiple environments.

**Comment by AliasIO** [comment]
Usually this is because of scripts that are loaded asynchronously and can be affected by network performance. After the page loads the script waits one second for any async scripts before completing the scan. Anything that loads later won't be included.

**Comment by kundan8239** [comment]
Is there any option to disable asynchronous behavior?

**Comment by AliasIO** [comment]
I'm not sure why you're getting Magento and Sitecore, here's what I get using the NPM driver:

```shell
node cli.js https://www.clairestudio.co.uk/ | jq .
```
```json
{
  "urls": {
    "https://www.clairestudio.co.uk/": {
      "status": 200
    }
  },
  "technologies": [
    {
      "slug": "varnish",
      "name": "Varnish",
      "description": "Varnish is a reverse caching proxy.",
      "confidence": 100,
      "version": null,
      "icon": "Varnish.svg",
      "website": "https://www.varnish-cache.org",
      "cpe": "cpe:2.3:a:varnish-software:varnish_cache:*:*:*:*:*:*:*:*",
      "categories": [
        {
          "id": 23,
          "slug": "caching",
          "name": "Caching"
        }
      ],
      "rootPath": true
    }
  ]
}
```

What is the full command you're using?

**Comment by kundan8239** [comment]
Hey @AliasIO,

### I've tried the three methods below and received different responses for each:

1. Using wappalyzer npm module (https://www.npmjs.com/package/wappalyzer)

```
const Wappalyzer = require('wappalyzer')
const options = {
  debug: false,
  delay: 500,
  headers: {},
  maxDepth: 3,
  maxWait: 20000,
  recursive: false,
  probe: true,
  proxy: false,
  noScripts: false,
  noRedirect: false,
  };

const wappalyzer = new Wappalyzer(options);
async function wappalyzerBase(url) {
  try {
    await wappalyzer.init()

    // Optionally set additional request headers
    //const headers = {}

    // Optionally set local and/or session storage
    // const storage = {
    //   local: {},
    //   session: {}
    // }

    const site = await wappalyzer.open(url)
    console.log("debug: Wappalyzer running for: "+url)
    // Optionally capture and output errors
    //site.on('error', console.error)

    const results = await site.analyze();
    console.log("debug: Got result from wappalyzer")
    return results
  } catch (error) {
    console.log("debug: wappalyzer error: "+ error)
    return Promise.reject(error);
  }
  finally {
    await wappalyzer.destroy()
  }
}
module.exports = wappalyzerBase
```

Response: 

![Screenshot 2023-08-04](https://github.com/wappalyzer/wappalyzer/assets/26499928/551893ef-78c4-4ca9-9d9f-9ab3c8368b6c)

2. Visiting [www.wappalyzer.com](http://www.wappalyzer.com/)

 Response: 

![Screenshot 2023-08-04 at 10 29 16 AM](https://github.com/wappalyzer/wappalyzer/assets/26499928/c6892057-9f52-45e4-8ccf-987b929c3e0f)

3.  Using cli:
 ```
  node src/drivers/npm/cli.js https://www.clairestudio.co.uk/
   ```
   Response: 
   ```
      {
        "urls": {
          "https://www.clairestudio.co.uk/": {
            "status": 200
          }
        },
        "technologies": [
          {
            "slug": "varnish",
            "name": "Varnish",
            "description": "Varnish is a reverse caching proxy.",
            "confidence": 100,
            "version": null,
            "icon": "Varnish.svg",
            "website": "https://www.varnish-cache.org",
            "cpe": "cpe:2.3:a:varnish-software:varnish_cache:*:*:*:*:*:*:*:*",
            "categories": [
              {
                "id": 23,
                "slug": "caching",
                "name": "Caching"
              }
            ],
            "rootPath": true
          }
        ]
      }
   ```
Can we increase async scripts' "wait time duration"  from 1 second to some other value, like 5 seconds? So that results can be consistent always. Is there any option available for the same or option `delay` will work?

**Comment by kala15** [comment]
> I'm not sure why you're getting Magento and Sitecore, here's what I get using the NPM driver:
> 
> ```shell
> node cli.js https://www.clairestudio.co.uk/ | jq .
> ```
> 
> ```json
> {
>   "urls": {
>     "https://www.clairestudio.co.uk/": {
>       "status": 200
>     }
>   },
>   "technologies": [
>     {
>       "slug": "varnish",
>       "name": "Varnish",
>       "description": "Varnish is a reverse caching proxy.",
>       "confidence": 100,
>       "version": null,
>       "icon": "Varnish.svg",
>       "website": "https://www.varnish-cache.org",
>       "cpe": "cpe:2.3:a:varnish-software:varnish_cache:*:*:*:*:*:*:*:*",
>       "categories": [
>         {
>           "id": 23,
>           "slug": "caching",
>           "name": "Caching"
>         }
>       ],
>       "rootPath": true
>     }
>   ]
> }
> ```
> 
> What is the full command you're using?

Below is option used

```
const options = {
  debug: false,
  delay: 100,
  headers: {},
  maxDepth: 3,
  maxWait: 5000,
  recursive: false,
  probe: true,
  proxy: false,
  noScripts: false,
  noRedirect: false,
  };
const wappalyzer = new Wappalyzer(options);
```
As per https://github.com/wappalyzer/wappalyzer/issues/7893#issuecomment-1664723731 comment, when probe is set to true tecnologies list is as mentioned below:

1. varnish
2. one.com
3. mysql
4. Magento
5. Microsoft ASP.NET
6. Sitecore
7. PHP

And magento and sitecore and their related technologies is being detected falsely.

original URL : https://clairestudio.co.uk/
redirected URL: https://www.designinfocus.co.uk/

Lets consider magento for example. based on `probe` option it gets appended to gets redirected URL inside the iframe tag like shown below

```
<!DOCTYPE html>
<html>
<head>
<title>"clairestudio.co.uk"</title>
<style type="text/css">
body {
	margin:0;
	padding:0;
}
iframe {
	display: block;
	border: none;
	height: 100vh;
	width: 100vw;
}
</style>
</head>
<body>
<iframe src= "https://www.designinfocus.co.uk/magento_version"><p>Your user agent does not support iframes. However you may visit <a href="https://www.designinfocus.co.uk/magento_version">the page that was supposed to be here</a></p></iframe>
</body>
</html>
```
as per line number https://github.com/wappalyzer/wappalyzer/blob/master/src/drivers/npm/driver.js#L1268

i.e. https://www.designinfocus.co.uk/magento_version which returns 404 error but still it will get executed as probe ok
at line https://github.com/wappalyzer/wappalyzer/blob/master/src/drivers/npm/driver.js#L1266 

when we scan https://www.designinfocus.co.uk/ throught npm magento and sitecore and their related technologies is not detected.

is it detecting falsely?

**Comment by kala15** [comment]
@AliasIO can you please help us to understand how **Magento and Sitecore** is being deteced as part of techology for https://www.clairestudio.co.uk/ using following option:
```
const options = {
  probe: true,
};

const wappalyzer = new Wappalyzer(options)
```

**Comment by AliasIO** [comment]
Right, it looks like the website returns a 200 and the 404 is embedded in an iframe. I made some small changes in 6fb1baedde9404fd60d220dcd0198ec39487e25b to check for specific content rather than just testing if the files exist.

**Comment by kundan8239** [comment]
> Right, it looks like the website returns a 200 and the 404 is embedded in an iframe. I made some small changes in [6fb1bae](https://github.com/wappalyzer/wappalyzer/commit/6fb1baedde9404fd60d220dcd0198ec39487e25b) to check for specific content rather than just testing if the files exist.
Once it is published on npm plz let us know I'll update version ..

**Comment by kundan8239** [comment]
Is this fix published?

**Comment by kala15** [comment]
@AliasIO The issue is fixed for `Sitecore` and not for `Magento` is still being detected as it may be  considering magento keyword in html content sample html content is given below:

```
<!DOCTYPE html>
<html>

<head>
	<title>"www.clairestudio.co.uk"</title>
	<style type="text/css">
		body {
			margin: 0;
			padding: 0;
		}

		iframe {
			display: block;
			border: none;
			height: 100vh;
			width: 100vw;
		}
	</style>
</head>

<body>
	<iframe src="https://www.designinfocus.co.uk/magento_version">
		<p>Your user agent does not support iframes. However you may visit <a
				href="https://www.designinfocus.co.uk/magento_version">the page that was supposed to be here</a></p>
	</iframe>
</body>
```
